units.md

```diff
@@ -0,0 +1,82 @@
+# Massa Units
+
+**Authors:**
+**Status:** Draft
+
+## Abstract
+
+This specification defines the units used to measure the different objects of the Massa blockchain. It aims to provide a standardized format for the different units to offer the simplest and most readable user experience possible while being consistent from one application to another within our ecosystem.
+
+> _NOTE:_ The manipulation of very large or very small numbers will be simplified by the use of [metric prefixes](https://en.wikipedia.org/wiki/Metric_prefix) of the [SI](https://en.wikipedia.org/wiki/International_System_of_Units).
+
+## Targeted Audience
+
+This specification is intended for developers who wish to interact with the Massa blockchain network using various tools and applications such as node API, smart contract tooling, dApps tooling, Thyra, and Thyra plugins.
+
+## Units
+
+### Massa coin
+
+Massa coin is the name of the native coin on the Massa blockchain.
+Massa coin is expressed as an unsigned 64-bit integer.
+
+1 Massa is equivalent to 1,000,000,000 nanoMassa, the smallest unit of Massa coin.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliMassa (mMASSA): 1 milliMassa represents 1,000,000 nanoMassa and also 1/1000th of a Massa.
+- microMassa (µMASSA): 1 microMassa represents 1,000 nanoMassa and also 1/1,000,000th of a Massa.
+- nanoMassa (nMASS): 1 nanoMassa represents the smallest unit of Massa coin and also 1/1,000,000,000th of a Massa.
+
+### Gaz
+
+Gaz is a virtual unit used to measure the computational complexity of executing transactions on the Massa blockchain. Each operation requires a certain amount of Gaz, and the amount of Gaz consumed is proportional to the computational complexity of the operation.
+Gaz is expressed as an unsigned 64-bit integer.
+
+1 Amphora is equivalent to 1,000,000,000 nanoAmphora, the smallest unit of gaz.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliAmphora (mAMPHORA): 1 milliAmphora represents 1,000,000 nanoAmphora or 1/1000th of an Amphora.
+- microAmphora (µAMPHORA): 1 microAmphora represents 1,000 nanoAmphora or 1/1,000,000th of an Amphora.
+- nanoAmphora (nAMPHORA): 1 nanoAmphora represents the smallest unit of gaz or 1/1,000,000,000th of an Amphora.
+
+### Roll
+
+Roll is a specific token used for staking on the Massa blockchain. A Roll token is required to create and endorse a block on the blockchain. The probability of being drawn to create or endorse a block is correlated to the number of rolls possed.
+Roll is expressed as an unsigned 64-bit integer.
+
+To simplify usage, the following standard units are recommended:
+
+kiloRoll (kROLL): 1 kiloRoll represents 1,000 rolls.
+megaRoll (MROLL): 1 megaRoll represents 1,000 kiloRolls or 1,000,000 rolls.
```

Comment:
1 roll = 100 massa
1 MAS = 0.01 ROLL
1mMAS = 0.00001 ROLL

why the units for rolls are kilo and mega and not milli, micro, nano. As far as I understand it would be easier because 1 ROLL is more than 1 MAS so if I have 1µMAS I can buy 0.01µROLL or 10nROLL

Reply:
I miss that you can't have less than one roll so 0.1 roll doesn't make sense.